### PR TITLE
Nft tracker

### PR DESCRIPTION
**Pull request by sneakzttv**
Added an NFT tracker to the web3auth GUI for token streaming.

image

image

Testing:
Log in with web3 auth.
Open the wallet.
Press the toggle button on custom tokens to see the Nft menu.
Add a 1155 NFT.
See NFT balance populate.

https://gyazo.com/1ed9cfe1ac3aa68eee75ec41792ae574

### CODE REVIEW

**Comment by sneakzttv** [comment]
Will complete docs once reviews are happy incase there is UI changes.

**Review by juans-chainsafe — COMMENTED** [review]
Should work with any ERC-1155? Im trying this NFT https://testnets.opensea.io/assets/sepolia/0xAdBa07EE1940f1dEC04b1981C3602EbFef012486/1
But 0 is displayed
Screenshot 2024-12-06 at 2 56 07 PM

Screenshot 2024-12-06 at 2 55 34 PM

**Comment by sneakzttv** [comment]
> Should work with any ERC-1155? Im trying this NFT https://testnets.opensea.io/assets/sepolia/0xAdBa07EE1940f1dEC04b1981C3602EbFef012486/1 But 0 is displayed Screenshot 2024-12-06 at 2 56 07 PM
> 
> Screenshot 2024-12-06 at 2 55 34 PM

yes, make sure your rpc is up to date with the current blocks

File: Packages/io.chainsafe.web3-unity.web3auth/Runtime/WalletGUI/Scripts/Web3AuthWalletGUITokenManager.cs
[the remark this answers is not in the document]
Comment (sneakzttv) [reply Packages/io.chainsafe.web3-unity.web3auth/Runtime/WalletGUI/Scripts/Web3AuthWalletGUITokenManager.cs:1]: sure, no problem.

[the remark this answers is not in the document]
Comment (sneakzttv) [reply Packages/io.chainsafe.web3-unity.web3auth/Runtime/WalletGUI/Scripts/Web3AuthWalletGUITokenManager.cs:1]: complete.

**Comment by kantagara** [comment]
> We should consider shifting test networks for efficiency. B3 is fast and easy to get testnet tokens, i vote there.
Yep I've already created the ticket for that 🫠

**Comment by kantagara** [comment]
> Should work with any ERC-1155?


Do you have it on your web3auth wallet?

File: Packages/io.chainsafe.web3-unity.web3auth/Runtime/WalletGUI/Scripts/Web3AuthCustomNftManager.cs
Comment (kantagara) [inline Packages/io.chainsafe.web3-unity.web3auth/Runtime/WalletGUI/Scripts/Web3AuthCustomNftManager.cs:1]: Much better :)

**Review by kantagara — APPROVED** [review]

**Review by rob1997 — APPROVED** [review]
kewl

**Comment by sneakzttv** [comment]
awaiting final check with updated RPC from @juans-chainsafe

**Review by juans-chainsafe — APPROVED** [review]
My bad, I was using a NFT minted in a different wallet than my w3a address.
I send a ERC1155 to my w3a address and worked!
Screenshot 2024-12-09 at 9 26 05 AM

**Pull request merged**